### PR TITLE
Type safe comparison in plugins - second iteration

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
Type safe comparison in plugins - second iteration
- some more string comparisons
- some bool
- some int

This PR is part of a set to try to separate some of the changes done in one of my previous batch PR's for the plugins directory, which is still on hold (#12228).
Once the new set is merged completely, it will hopefully reduce the changes in that PR, so it can be reviewed easier and finally be merged.

The changes in this PR should be also be fairly easy to review. In hope that  this will get merged quickly. ;)

Note: Don't bother if some possible changes are missing or could be differently written. They are probably in the batch PR , that this one references. As soon as this set of sub PR's is merged, the batch PR will have its conflicts resolved and should be a lot easier to review and finally get merged.

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by wilsonge** [comment]
Conflicts with the other PR :)

**Comment by andrepereiradasilva** [comment]
I think you need to review the strict comparations ... some are not correct



For instance:
https://github.com/joomla/joomla-cms/pull/13272/files#diff-72a6846b2b3372956e666bfffccd2ed7R99

https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/language/multilang.php#L27

**Comment by frankmayer** [comment]
@andrepereiradasilva Thanks missed the one you mentioned and also reversed another case that has multiple comparisons. In that second case though, there might be some cleaning up to do in future (false/null is used if no language is given, though maybe it could just be a '')

File: plugins/content/vote/vote.php [hunk 1]
```diff
@@ -128,7 +128,7 @@ private function displayVotingData($context, &$row, &$params, $page)
 		include $path;
 		$html = ob_get_clean();
 
-		if ($this->app->input->getString('view', '') === 'article' && $row->state == 1)
+		if ($this->app->input->getString('view', '') === 'article' && $row->state === 1)
```
Comment (andrepereiradasilva) [inline plugins/content/vote/vote.php]: doubt: is row state (and access in changes below) an int for sure in all cases?

**Comment by frankmayer** [comment]
It would be great, if this sub-PR could be reviewed / tested and merged, so that the associated Batch PR can also be easier reviewed / tested and merged.

BTW, :+1: to all for putting in the extra work to support my efforts for all these PR with lots and lots of changes.

**Comment by andrepereiradasilva** [comment]
> BTW, 👍 to all for putting in the extra work to support my efforts for all these PR with lots and lots of changes.

no, thank you for doing this!

File: plugins/editors-xtd/image/image.php [hunk 2]
```diff
@@ -40,15 +40,15 @@ public function onDisplay($name, $asset, $author)
...
 			$asset = $extension;
 		}
 
 		if ($user->authorise('core.edit', $asset)
 			|| $user->authorise('core.create', $asset)
 			|| (count($user->getAuthorisedCategories($asset, 'core.create')) > 0)
-			|| ($user->authorise('core.edit.own', $asset) && $author == $user->id)
+			|| ($user->authorise('core.edit.own', $asset) && $author === $user->id)
 			|| (count($user->getAuthorisedCategories($extension, 'core.edit')) > 0)
 			|| (count($user->getAuthorisedCategories($extension, 'core.edit.own')) > 0 && $author == $user->id))
```
Comment (shur) [inline plugins/editors-xtd/image/image.php]: and what about line 53 ?

Comment (frankmayer) [inline plugins/editors-xtd/image/image.php]: Ah, yes, missed that one... Thanks!!

File: plugins/content/vote/vote.php (hunk 1 above)
Comment (andrepereiradasilva) [inline plugins/content/vote/vote.php]: Did you check this?

Comment (frankmayer) [inline plugins/content/vote/vote.php]: No, sorry, it got under. I checked it just now. You were right! Unfortunately it is not guaranteed that it is an int. 
So, either we cast it to an int or we revert the change.

Comment (frankmayer) [inline plugins/content/vote/vote.php]: BTW, I don't know if this has come up in the past, but if we want to do strict(er) typing, as we should, then I guess the db driver classes should make sure the right types are returned, also.

Comment (andrepereiradasilva) [inline plugins/content/vote/vote.php]: i agree, but a change like that IMHO should be in 4.0 branch
@wilsonge what do you think?

**Comment by andrepereiradasilva** [comment]
@frankmayer the one i pointed and you corrected is not the only case in this PR. (example: access)
Please review the PR changes according to that. IMO i simportant to make sure this PR don't break anything by making incorrect assumptions reggarding the variable type.

**Comment by frankmayer** [comment]
@andrepereiradasilva Yes, I was already checking those out, too...
Commit will follow soon.

**Comment by andrepereiradasilva** [comment]
ok sorry, tought you missed them. ignore the comment them

**Comment by frankmayer** [comment]
No, no, my fault. I shouldn't have pushed the commit, before completing my checks. I was multitasking with other things... :smile_cat:

**Comment by frankmayer** [comment]
The rest should be OK now.

**Comment by frankmayer** [comment]
It would be great, if this sub-PR could be reviewed / tested and merged, so that the referenced Batch PR #12228 can also be easier reviewed / tested and merged.

**Comment by frankmayer** [comment]
Conflicts resolved...

**Comment by frankmayer** [comment]
Pls check and merge.

File: plugins/content/pagebreak/pagebreak.php [hunk 3]
```diff
@@ -311,17 +311,17 @@ protected function _createToc(&$row, &$matches, &$page)
 				$title = JText::sprintf('PLG_CONTENT_PAGEBREAK_PAGE_NUM', $i);
 			}
 
-			$liClass   = ($limitstart == $i - 1) ? ' class="active"' : '';
-			$class     = ($limitstart == $i - 1) ? 'toclink active' : 'toclink';
+			$liClass   = ($limitstart === $i - 1) ? ' class="active"' : '';
+			$class     = ($limitstart === $i - 1) ? 'toclink active' : 'toclink';
```
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php]: Remove parentheses.

File: plugins/content/pagebreak/pagebreak.php [hunk 4]
```diff
@@ -311,17 +311,17 @@ protected function _createToc(&$row, &$matches, &$page)
...
 			$i++;
 		}
 
 		if ($this->params->get('showall'))
 		{
 			$link      = JRoute::_(ContentHelperRoute::getArticleRoute($row->slug, $row->catid, $row->language) . '&showall=1&limitstart=');
-			$liClass   = ($limitstart == $i - 1) ? ' class="active"' : '';
-			$class     = ($limitstart == $i - 1) ? 'toclink active' : 'toclink';
+			$liClass   = ($limitstart === $i - 1) ? ' class="active"' : '';
+			$class     = ($limitstart === $i - 1) ? 'toclink active' : 'toclink';
```
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php]: Remove parentheses.

File: plugins/editors-xtd/image/image.php (hunk 2 above)
Comment (Quy) [inline plugins/editors-xtd/image/image.php]: Remove parentheses.

File: plugins/search/tags/tags.php [hunk 5]
```diff
@@ -156,7 +156,7 @@ public function onContentSearch($text, $phrase = '', $ordering = '', $areas = nu
 			foreach ($rows as $key => $row)
 			{
 				$rows[$key]->href       = TagsHelperRoute::getTagRoute($row->id);
-				$rows[$key]->text       = ($row->description != '' ? $row->description : $row->title);
+				$rows[$key]->text       = ($row->description !== '' ? $row->description : $row->title);
```
Comment (Quy) [inline plugins/search/tags/tags.php]: plugins/editors-xtd/image/image.php

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 572572cb9c6bdca19b9ba13119d51b3b7f2187df<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13272">issues.joomla.org/tracker/joomla-cms/13272</a>.</sub>

**Comment by frankmayer** [comment]
Thanks, one more tester to get this RTC?

File: plugins/captcha/recaptcha/recaptchalib.php [hunk 6]
```diff
@@ -53,7 +53,7 @@ class JReCaptcha
 	 */
 	public function __construct($secret)
 	{
-		if ($secret == null || $secret == '')
+		if ($secret == null || $secret === '')
```
Comment (Quy) [inline plugins/captcha/recaptcha/recaptchalib.php]: @frankmayer Should this be `=== null`?

Comment (frankmayer) [inline plugins/captcha/recaptcha/recaptchalib.php]: I think It could, and I also think I have done this in another PR.
